src/Microsoft.ML.CpuMath/AvxIntrinsics.cs

```diff
@@ -425,7 +425,7 @@ public static unsafe void AddScalarU(float scalar, Span<float> dst)
 
                 Vector256<float> scalarVector256 = Vector256.Create(scalar);
 
-                while (pDstCurrent + 8 <= pDstEnd)
+                while (pDstCurrent <= pDstEnd - 8)
```

Comment:
nit: Explicitly cache the loop-end in a local to ensure it also isn't recomputed?

Reply:
fixed